tokens/src/commands.rs

```diff
@@ -990,4 +1048,309 @@ mod tests {
             panic!("expected SetLockup instruction");
         }
     }
+
+    fn tmp_file_path(name: &str, pubkey: &Pubkey) -> String {
+        use std::env;
+        let out_dir = env::var("FARF_DIR").unwrap_or_else(|_| "farf".to_string());
+
+        format!("{}/tmp/{}-{}", out_dir, name, pubkey.to_string())
+    }
+
+    #[test]
+    fn test_check_payer_balances() {
+        let (mut genesis_config, sender_keypair) =
+            create_genesis_config(sol_to_lamports(9_000_000.0));
+        genesis_config.fee_rate_governor = FeeRateGovernor::new(10_000, 0);
+        let bank_forks = Arc::new(RwLock::new(BankForks::new(Bank::new(&genesis_config))));
+        Runtime::new().unwrap().block_on(async {
+            let transport = start_local_server(&bank_forks).await;
+            let mut banks_client = start_client(transport).await.unwrap();
+
+            let sender_keypair_file = tmp_file_path("keypair_file", &sender_keypair.pubkey());
+            write_keypair_file(&sender_keypair, &sender_keypair_file).unwrap();
+
+            let recipient = Pubkey::new_rand();
+            let allocations = vec![Allocation {
+                recipient: recipient.to_string(),
+                amount: 1000.0,
+                lockup_date: "".to_string(),
+            }];
+
+            // Test distribute tokens with fully funded single payer
+            let mut args = DistributeTokensArgs {
+                sender_keypair: read_keypair_file(&sender_keypair_file).unwrap().into(),
```

Comment:
What's with all the files?

Reply:
Easiest way I know of to be able to use the same Keypair in more than one signing role, since Keypair doesn't support clone. Open to another pattern, if you have a suggestion.